CC26xx: increase size of the region reserved for stack from 256 bytes to 1024 bytes

atiselsts opened the pull request:
As previously discussed in the Contiki issue tracker, the size of memory reserved for stack region on CC26xx is inadequately small.

For example, the cc26xx-web-demo to process some simple CoAP requests use 768 bytes of stack (measured with the stack check library). It has >1000 bytes of stack memory actually available, so it doesn't crash, but this is still bad. Custom applications developed on top of this demo may compile and link, give their developers false sense of security, but then fail at runtime.

This PR proposes to increase it to 1024 bytes (In SPHERE, we're using ~2000 bytes). With the new "build all" script I tested that this setting does not cause build failures on any examples - all examples that build with 256 byte setting also build with 1024 byte setting.

g-oikonomou commented:
Makes sense, but the web demo build fails on travis?

atiselsts commented:
So it actually fails on sensortag if the stack region size in the linker script is set to 600 bytes or more. Which means that there's already a real problem.

It builds ok on `BOARD=launchpad/cc2650` and `BOARD=srf06/cc26xx` - that's what I checked before.
I need to add also building for all boards in my "build all" script, not just for all platforms.

g-oikonomou commented:
When I was writing the porting guide the other day, I was thinking that we should perhaps get rid of `TARGET` and `BOARD` and switch to a `TARGET=srf06-cc26xx/sensortag/cc2650` model, if you see what I mean.

atiselsts commented:
This builds for all boards as well as all platforms. Takes ~15 min on the server.

**Question:** what to do with the problems found as the result of this? Create a number of issues and hope that maintainers will pick fix them? Alternatively, investigate and propose pull requests.

The CI tests fail, but not because of new bugs introduced, but rather existing bugs uncovered: `6p-packet` never could be built on `zoul`, because this platforms is missing `node_id`; it's just that previously the `compile-arm-ports-02` CI test was not actually building it.

Docker is super fast and convenient to set up!

Results:

> Number of examples skipped or built successfully: 738
> Number of examples that failed to build: 126
> Failed examples: ; ../../examples/6tisch/6p-packet for cc2538dk (default); ../../examples/ip64-router for cc2538dk (default); ../../examples/platform-specific/cc2538-common/mqtt-demo for cc2538dk (default); ../../examples/ipv6-hooks for cc2538dk (default); ../../examples/storage/antelope-shell for cc2538dk (default); ../../examples/ip64-router for native (default); ../../examples/slip-radio for native (default); ../../examples/dev/gpio-hal for native (default); ../../examples/libs/stack-check for native (default); ../../examples/ipv6-hooks for native (default); ../../examples/storage/antelope-shell for native (default); ../../examples/storage/cfs-coffee for native (default); ../../examples/rpl-udp for nrf52dk (default); ../../examples/rpl-border-router for nrf52dk (default); ../../examples/ip64-router for nrf52dk (default); ../../examples/http-socket for nrf52dk (default); ../../examples/multicast for nrf52dk (default); ../../examples/slip-radio for nrf52dk (default); ../../examples/dev/gpio-hal for nrf52dk (default); ../../examples/platform-specific/nrf52dk/mqtt-demo for nrf52dk (default); ../../examples/platform-specific/nrf52dk/coap-demo/coap-server for nrf52dk (default); ../../examples/platform-specific/nrf52dk/coap-demo/coap-client for nrf52dk (default); ../../examples/ipv6-hooks for nrf52dk (default); ../../examples/storage/antelope-shell for nrf52dk (default); ../../examples/storage/cfs-coffee for nrf52dk (default); ../../examples/coap for nrf52dk (default); ../../examples/6tisch/6p-packet for openmote-cc2538 (default); ../../examples/ip64-router for openmote-cc2538 (default); ../../examples/platform-specific/cc2538-common/mqtt-demo for openmote-cc2538 (default); ../../examples/ipv6-hooks for openmote-cc2538 (default); ../../examples/storage/antelope-shell for openmote-cc2538 (default); ../../examples/ip64-router for sky (default); ../../examples/http-socket for sky (default); ../../examples/ipso-objects for sky (default); ../../examples/dev/gpio-hal for sky (default); ../../examples/libs/logging for sky (default); ../../examples/ipv6-hooks for sky (default); ../../examples/storage/antelope-shell for sky (default); ../../examples/6tisch/etsi-plugtest-2017 for srf06-cc26xx (sensortag/cc2650); ../../examples/6tisch/etsi-plugtest-2017 for srf06-cc26xx (sensortag/cc1350); ../../examples/ip64-router for srf06-cc26xx (srf06/cc26xx); ../../examples/ip64-router for srf06-cc26xx (srf06/cc13xx); ../../examples/ip64-router for srf06-cc26xx (launchpad/cc2650); ../../examples/ip64-router for srf06-cc26xx (launchpad/cc1350); ../../examples/ip64-router for srf06-cc26xx (sensortag/cc2650); ../../examples/ip64-router for srf06-cc26xx (sensortag/cc1350); ../../examples/ipso-objects for srf06-cc26xx (sensortag/cc2650); ../../examples/ipso-objects for srf06-cc26xx (sensortag/cc1350); ../../examples/ipv6-hooks for srf06-cc26xx (srf06/cc26xx); ../../examples/ipv6-hooks for srf06-cc26xx (srf06/cc13xx); ../../examples/ipv6-hooks for srf06-cc26xx (launchpad/cc2650); ../../examples/ipv6-hooks for srf06-cc26xx (launchpad/cc1350); ../../examples/ipv6-hooks for srf06-cc26xx (sensortag/cc2650); ../../examples/ipv6-hooks for srf06-cc26xx (sensortag/cc1350); ../../examples/storage/antelope-shell for srf06-cc26xx (srf06/cc26xx); ../../examples/storage/antelope-shell for srf06-cc26xx (srf06/cc13xx); ../../examples/storage/antelope-shell for srf06-cc26xx (launchpad/cc2650); ../../examples/storage/antelope-shell for srf06-cc26xx (launchpad/cc1350); ../../examples/storage/antelope-shell for srf06-cc26xx (sensortag/cc2650); ../../examples/storage/antelope-shell for srf06-cc26xx (sensortag/cc1350); ../../examples/storage/cfs-coffee for srf06-cc26xx (srf06/cc26xx); ../../examples/storage/cfs-coffee for srf06-cc26xx (srf06/cc13xx); ../../examples/storage/cfs-coffee for srf06-cc26xx (launchpad/cc2650); ../../examples/storage/cfs-coffee for srf06-cc26xx (launchpad/cc1350); ../../examples/storage/cfs-coffee for srf06-cc26xx (sensortag/cc2650); ../../examples/storage/cfs-coffee for srf06-cc26xx (sensortag/cc1350); ../../examples/6tisch/sixtop for zoul (orion); ../../examples/6tisch/etsi-plugtest-2017 for zoul (orion); ../../examples/6tisch/6p-packet for zoul (remote-reva); ../../examples/6tisch/6p-packet for zoul (remote-revb); ../../examples/6tisch/6p-packet for zoul (firefly-reva); ../../examples/6tisch/6p-packet for zoul (firefly); ../../examples/6tisch/6p-packet for zoul (orion); ../../examples/6tisch/simple-node for zoul (orion); ../../examples/rpl-border-router for zoul (orion); ../../examples/ip64-router for zoul (remote-reva); ../../examples/ip64-router for zoul (remote-revb); ../../examples/ip64-router for zoul (firefly-reva); ../../examples/ip64-router for zoul (firefly); ../../examples/nullnet for zoul (orion); ../../examples/http-socket for zoul (orion); ../../examples/multicast for zoul (remote-reva); ../../examples/multicast for zoul (remote-revb); ../../examples/multicast for zoul (firefly-reva); ../../examples/multicast for zoul (firefly); ../../examples/multicast for zoul (orion); ../../examples/platform-specific/zoul/at-test for zoul (remote-reva); ../../examples/platform-specific/zoul/at-test for zoul (remote-revb); ../../examples/platform-specific/zoul/at-test for zoul (firefly-reva); ../../examples/platform-specific/zoul/at-test for zoul (firefly); ../../examples/platform-specific/zoul/at-test for zoul (orion); ../../examples/platform-specific/zoul/rtcc for zoul (remote-reva); ../../examples/platform-specific/zoul/rtcc for zoul (remote-revb); ../../examples/platform-specific/zoul/rtcc for zoul (firefly-reva); ../../examples/platform-specific/zoul/rtcc for zoul (firefly); ../../examples/platform-specific/zoul/rtcc for zoul (orion); ../../examples/platform-specific/zoul/orion/client for zoul (remote-reva); ../../examples/platform-specific/zoul/orion/client for zoul (remote-revb); ../../examples/platform-specific/zoul/orion/client for zoul (firefly-reva); ../../examples/platform-specific/zoul/orion/client for zoul (firefly); ../../examples/platform-specific/zoul/orion/client for zoul (orion); ../../examples/platform-specific/zoul/orion/ip64-router for zoul (remote-reva); ../../examples/platform-specific/zoul/orion/ip64-router for zoul (remote-revb); ../../examples/platform-specific/zoul/orion/ip64-router for zoul (firefly-reva); ../../examples/platform-specific/zoul/orion/ip64-router for zoul (firefly); ../../examples/platform-specific/zoul/rev-b for zoul (remote-reva); ../../examples/platform-specific/zoul/rev-b for zoul (remote-revb); ../../examples/platform-specific/zoul/rev-b for zoul (firefly-reva); ../../examples/platform-specific/zoul/rev-b for zoul (firefly); ../../examples/platform-specific/zoul/rev-b for zoul (orion); ../../examples/platform-specific/cc2538-common/mqtt-demo for zoul (remote-reva); ../../examples/platform-specific/cc2538-common/mqtt-demo for zoul (remote-revb); ../../examples/platform-specific/cc2538-common/mqtt-demo for zoul (firefly-reva); ../../examples/platform-specific/cc2538-common/mqtt-demo for zoul (firefly); ../../examples/platform-specific/cc2538-common/mqtt-demo for zoul (orion); ../../examples/ipv6-hooks for zoul (remote-reva); ../../examples/ipv6-hooks for zoul (remote-revb); ../../examples/ipv6-hooks for zoul (firefly-reva); ../../examples/ipv6-hooks for zoul (firefly); ../../examples/ipv6-hooks for zoul (orion); ../../examples/storage/antelope-shell for zoul (remote-reva); ../../examples/storage/antelope-shell for zoul (remote-revb); ../../examples/storage/antelope-shell for zoul (firefly-reva); ../../examples/storage/antelope-shell for zoul (firefly); ../../examples/storage/antelope-shell for zoul (orion); ../../examples/coap for zoul (orion)

atiselsts commented:
Could it be possible to reduce some config parameters or disable some functionality by default on SensorTag to avoid the memory overruns. One way is to reduce `UIP_CONF_BUFFER_SIZE` in project-conf.h from 900 to something smaller, like 400. Can you comment on this, George?

(Not sure what `NETSTCK_ROUTING_STATE_SIZE` is in the config, but it looks misspelled.)

With the new build all script I can check that all platform-specific examples are building afterwards:

   $ PLATFORMS=srf06-cc26xx MAKEFILES=`find ../../examples/platform-specific/ -name Makefile` ./build.sh

atiselsts commented:
@g-oikonomou comments?

@simonduq the `NETSTCK_ROUTING_STATE_SIZE` name was also mentioned in Gitter by someone, any comments on what it is meant to be?

g-oikonomou commented:
Sorry I missed your asking last time round. We should come up with a way to reduce things so it fits on the sensortag too. Queuebuf reduction could also help

simonduq commented:
@atiselsts `NETSTCK_ROUTING_STATE_SIZE` should be replaced with `NETSTACK_MAX_ROUTE_ENTRIES`

@g-oikonomou regarding platform/board, just wanted to remind you of https://github.com/contiki-ng/contiki-ng/issues/83

g-oikonomou commented:
Not sure what #83 has to do?

simonduq commented:
You posted earlier on the use of `TARGET` and `BOARD`

g-oikonomou commented:
Ja. Here I was referring to getting rid of the concept of `BOARD` (or of `TARGET`) altogether. #83 is about renaming the platform, right? (But we digress... :D)

atiselsts commented:
Updated the `UIP_CONF_BUFFER_SIZE` to 500 and changed the name of the other thing  to `NETSTACK_MAX_ROUTE_ENTRIES`. Not sure if there are any specific tests in which the smaller buffer size could make a difference? (What I mean is "does this have to be tested in a particular way?")

g-oikonomou commented:
Even though they are pretty long, 500 should be enough for the sensortag MQTT publish messages. I say let's merge this.
